app/views/RoomView/index.tsx

```diff
@@ -347,12 +351,19 @@ class RoomView extends React.Component<IRoomViewProps, IRoomViewState> {
 
 	shouldComponentUpdate(nextProps: IRoomViewProps, nextState: IRoomViewState) {
 		const { state } = this;
-		const { roomUpdate, member, isOnHold } = state;
+		const { roomUpdate, member, isOnHold, showAnnouncementModal } = state;
 		const { appState, theme, insets, route } = this.props;
 		if (theme !== nextProps.theme) {
 			return true;
 		}
 		if (appState !== nextProps.appState) {
+			if (appState === 'background' && showAnnouncementModal) {
```

Comment:
We shouldn't do this.
It's going to be a mess to remove it later.

What if you just hide the modal before going background?